### DISCUSSION TITLE
Unnecessary rendering

### DISCUSSION THREAD

**Discussion by sftechswdev** [jpudysz/react-native-unistyles#169] [Q&A] [answered]
Hi,
Upon integrating the react-native-unistyles library into my project and configuring it according to the provided instructions, I've noticed that the component I'm working with is rendering twice. Is this expected behaviour?  Please use the sample code below to check it.

Version:
"react-native": "0.72.12",
"react-native-unistyles": "^2.5.1"

App.tsx
```
import React from 'react';
import {SafeAreaView, Text} from 'react-native';
import './unistyles';
import {createStyleSheet, useStyles} from 'react-native-unistyles';

function App(): JSX.Element {
  const {styles} = useStyles(styleSheet);

  console.log('render count');

  return (
    <SafeAreaView style={styles.appContainer}>
      <Text style={styles.text}>Hello World!</Text>
    </SafeAreaView>
  );
}

const styleSheet = createStyleSheet(theme => ({
  appContainer: {
    flex: 1,
    backgroundColor: theme.colors.background,
  },
  text: {
    fontSize: 25,
    color: theme.colors.text,
  },
}));

export default App;
```

unistyles.ts
```
import {UnistylesRegistry} from 'react-native-unistyles';

const lightTheme = {
  colors: {
    background: 'rgb(200, 200, 200)',
    text: 'rgb(25, 25, 25)',
  },
};

type AppThemes = {
  light: typeof lightTheme;
};

declare module 'react-native-unistyles' {
  export interface UnistylesThemes extends AppThemes {}
}

UnistylesRegistry.addThemes({
  light: lightTheme,
}).addConfig({
  adaptiveThemes: false,
});
```

**Comment by jpudysz** [comment]
Copy-pasted your example and here is the result:

![image](https://github.com/jpudysz/react-native-unistyles/assets/9088288/304bfda4-d488-408f-bca3-7724240d0389)

**Reply by sftechswdev** [reply]
Thanks for the quick response. I'm new to React and must be doing something wrong. This happens on Android emulator and can't confirm if it also happens on iOS emulator. 
![image](https://github.com/jpudysz/react-native-unistyles/assets/129957684/74bf14e1-1cbd-489d-9856-fc00e8f0c6f0)

**Reply by jpudysz** [reply]
I will check it on Android too, will let you know

**Reply by sftechswdev** [reply]
I can confirm that it doesn't happen on iOS.
![image](https://github.com/jpudysz/react-native-unistyles/assets/129957684/a332eee2-0422-4d37-afc3-5caaf6825fe0)

**Comment by jpudysz** [comment]
On Android is the same behaviour (one single render of the App component). I don't get it, you have the same output in the console. Where is the problem?

![android](https://github.com/jpudysz/react-native-unistyles/assets/9088288/809f2d69-ddd4-490e-8917-33dcd95b5d23)

**Reply by rikur** [reply]
Try without SafeAreaView. Just an idea.

**Reply by sftechswdev** [reply]
Hi Riku,
Replacing SafeAreaView with View didn't help.

**Reply by sftechswdev** [reply]
I've pushed the sample project to https://github.com/sftechswdev/rnunistyles. Could you please check if it has anything to do with the setup of the project.

**Reply by jpudysz** [reply]
I've run your repository, and it looks exactly the same on Android:

![android](https://github.com/jpudysz/react-native-unistyles/assets/9088288/b5ab06b9-e9d4-42a9-839d-bbca55b9cd4c)

**Reply by sftechswdev** [reply]
I noticed something with the insets. In the first render, top and bottom has values and 0 in the second render.  

https://github.com/jpudysz/react-native-unistyles/assets/129957684/a231a24a-f43f-4402-99d0-30bdbdbb66f8

**Reply by jpudysz** [reply]
Oh i see! You're using simulator with the notch. Nice catch!

Here is the simillar issue: https://github.com/jpudysz/react-native-unistyles/issues/170

I will work on it this week

**Reply by sftechswdev** [reply]
Thanks a lot.

**Reply by jpudysz** [reply, marked as answer]
No, thank you a lot!

Please check new version 2.5.2 with fix.

https://github.com/jpudysz/react-native-unistyles/releases/tag/v2.5.2

**Reply by sftechswdev** [reply]
Resolved, Thanks for the quick fix.
